### PR TITLE
MM-20696: Use SAML Identity Provider Metadata for SAML setup

### PR DESCRIPTION
**Pull request by catalintomai**
#### Summary
Use SAML Identity Provider Metadata for SAML setup

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-20696

Parent Ticket: https://mattermost.atlassian.net/browse/MM-19771

#### Related Pull Requests
- Has webapp changes : https://github.com/mattermost/mattermost-webapp/pull/4427
- Has server changes : https://github.com/mattermost/mattermost-server/pull/13311
- Has enterprise changes : https://github.com/mattermost/enterprise/pull/545

### CODE REVIEW

File: src/action_types/admin.ts [hunk 1]
```diff
@@ -184,4 +184,12 @@ export default keyMirror({
...
     ENABLED_PLUGIN: null,
     DISABLED_PLUGIN: null,
+
+    SAML_GET_METADATA_REQUEST: null,
+    SAML_GET_METADATA_SUCCESS: null,
+    SAML_GET_METADATA_FAILURE: null,
+
+    SET_SAML_IDP_REQUEST: null,
+    SET_SAML_IDP_SUCCESS: null,
+    SET_SAML_IDP_FAILURE: null,
```
Comment (enahum) [inline src/action_types/admin.ts]: @catalintomai I know I have not been assigned as a reviewer, but wanted to ask if you need to keep track of the requests? If not perhaps best to remove this

Comment (mkraft) [reply src/action_types/admin.ts]: @catalintomai To elaborate on this, the convention used to be to emit an action for every request, success, and failure. However, it seems that now [unused actions are being removed](https://github.com/mattermost/mattermost-redux/pull/992/files) from the repository.

Comment (catalintomai) [reply src/action_types/admin.ts]: Thanks, the spurious requests should be cleaned up now.

**Comment by srkgupta** [comment]
Hi @catalintomai 
I am trying to test all the related PRs together, but I get the error as shown in the screenshot. I had merged all the PR's code into my local repos but looks like I am either missing something or something's wrong. Can you please check. 

PRs being tested together locally:
https://github.com/mattermost/mattermost-redux/pull/1003
https://github.com/mattermost/mattermost-server/pull/13311
https://github.com/mattermost/mattermost-webapp/pull/4427

JS error while trying to make request:
Screenshot 2019-12-26 at 7 18 18 PM

**Comment by catalintomai** [comment]
> Hi @catalintomai
> I am trying to test all the related PRs together, but I get the error as shown in the screenshot. I had merged all the PR's code into my local repos but looks like I am either missing something or something's wrong. Can you please check.
> 
> PRs being tested together locally:
> #1003
> [mattermost/mattermost-server#13311](https://github.com/mattermost/mattermost-server/pull/13311)
> [mattermost/mattermost-webapp#4427](https://github.com/mattermost/mattermost-webapp/pull/4427)
> 
> JS error while trying to make request:
> Screenshot 2019-12-26 at 7 18 18 PM

Hi Rohitesh,
I merged master into the PRS above to make sure we are dealing with the latest. The error above one usually gets if webapp doesn't reference the latest redux changes. Have you run "npm run dev" in the redux directory before doing a make-run in the server directory?

**Pull request merged**